README.md

```diff
@@ -8,14 +8,157 @@ Based on steverhoades/oauth2-openid-connect-client.
+
+## Documentation
+
+### Tokens
+
+You can use authenticated user object to get tokens.
+
+```php
+// Get login user, returns \LaravelOIDCAuth\Contracts\OIDCAuthenticatable|null
+Auth::user();
+// Get the saved access token, returns \OpenIDConnectClient\AccessToken|null
+Auth::user()->accessToken();
+// Get the saved id token, returns \Lcobucci\JWT\Token|null
+Auth::user()->accessToken()->getIdToken();
+// Get the saved refresh token, returns string|null
+Auth::user()->accessToken()->getRefreshToken();
+```
+
+### ID Token
+
+You can get id token claims like this:
+
+```php
+Route::get('/user', function () {
+    return Auth::user()->accessToken()->getIdToken()->getClaims();
+});
+```
+
+### Refresh the access token
+
+If you are using `oidc-auth-session` guard with `OIDCUser`, you can easily refresh the access token:
+
+```php
+use LaravelOIDCAuth\Facades\OIDC;
+// ...
+Auth::user()->refreshAccessToken();
+```
+
+Typically, you don't need to refresh the access token manually, the following doc shows an automatic way to do this.
+
+#### Auto refresh
+
+Sometimes, you want the `oidc-auth-session` to refresh the access token automatically every request, please edit
+`config/oidc_auth.php`:
+
+```php
+    // ...
+    'auto_refresh' => true,
+    // ...
+```
+
+Next time the user logged in, the `CallbackController` will set `rememberMe` option from
+`config('oidc_auth.auto_refresh')` to the guard's `login` function.
+
+### User Model & User Factory
+
+If you need an authenticated user to be an eloquent model with OIDC features, your user model must implement
+`LaravelOIDCAuth\Contracts\OIDCAuthenticatable`:
+
+```php
+namespace App\Models;
+
+use Illuminate\Notifications\Notifiable;
+use Illuminate\Contracts\Auth\MustVerifyEmail;
+use LaravelOIDCAuth\Contracts\OIDCAuthenticatable;
+use LaravelOIDCAuth\HasAccessToken;
+use LaravelOIDCAuth\RefreshesAccessToken;
+use Illuminate\Foundation\Auth\User as Authenticatable;use OpenIDConnectClient\AccessToken;
+
+class User extends Authenticatable implements OIDCAuthenticatable
+{
+    use Notifiable, HasAccessToken, RefreshesAccessToken;
+
+    // ...
+}
+```
+
+Two trait `HasAccessToken`, `RefreshesAccessToken` are added.
+
+Next, you can implement a user factory `LaravelOIDCAuth\Contracts\OIDCAuthenticatableFactory` to get user from database:
+
+```php
+namespace App;
+
+use LaravelOIDCAuth\Contracts\OIDCAuthenticatable;
+use LaravelOIDCAuth\Contracts\OIDCAuthenticatableFactory;
+use OpenIDConnectClient\AccessToken;
+
+class UserFactory implements OIDCAuthenticatableFactory
+{
+    public function authenticatable(AccessToken $token): OIDCAuthenticatable
+    {
+        $user = User::firstOrCreate(/* ... */);
+        $user->setAccessToken($token);
+        return $user;
+    }
+}
+```
+
+### Customization
+
+#### Handle OIDC callback
+
+When the OIDC Provider authenticated the user, the browser will be redirected to the callback route on your own web
```

Comment:
When user authenticates via OIDC,

Reply:
(I don't really think you should explain callback here. Consumer should already know it if they need to handle that manually)

Reply:
Okay.